component/otelcol/exporter/loadbalancing/loadbalancing.go

```diff
@@ -0,0 +1,246 @@
+	// However, the block should not be "optional" so that the defaults are populated.
+	Client GRPCClientArguments `river:"client,block"`
+}
+
+func (otlpConfig OtlpConfig) Convert() otlpexporter.Config {
+	return otlpexporter.Config{
+		TimeoutSettings: exporterhelper.TimeoutSettings{
+			Timeout: otlpConfig.Timeout,
+		},
+		QueueSettings:      *otlpConfig.Queue.Convert(),
+		RetrySettings:      *otlpConfig.Retry.Convert(),
+		GRPCClientSettings: *otlpConfig.Client.Convert(),
+	}
+}
+
+// ResolverSettings defines the configurations for the backend resolver
+type ResolverSettings struct {
+	Static *StaticResolver `river:"static,block,optional"`
+	DNS    *DNSResolver    `river:"dns,block,optional"`
+}
+
+func (resolverSettings ResolverSettings) Convert() loadbalancingexporter.ResolverSettings {
+	res := loadbalancingexporter.ResolverSettings{}
+
+	if resolverSettings.Static != nil {
+		staticResolver := resolverSettings.Static.Convert()
+		res.Static = &staticResolver
+	}
+
+	if resolverSettings.DNS != nil {
+		dnsResover := resolverSettings.DNS.Convert()
+		res.DNS = &dnsResover
+	}
+
+	return res
+}
+
+// StaticResolver defines the configuration for the resolver providing a fixed list of backends
+type StaticResolver struct {
+	Hostnames []string `river:"hostnames,attr"`
+}
+
+func (staticResolver StaticResolver) Convert() loadbalancingexporter.StaticResolver {
+	return loadbalancingexporter.StaticResolver{
+		Hostnames: staticResolver.Hostnames,
+	}
+}
+
+// DNSResolver defines the configuration for the DNS resolver
+type DNSResolver struct {
+	Hostname string        `river:"hostname,attr"`
+	Port     string        `river:"port,attr,optional"`
+	Interval time.Duration `river:"interval,attr,optional"`
+	Timeout  time.Duration `river:"timeout,attr,optional"`
+}
+
+var (
+	_ river.Defaulter = &DNSResolver{}
+)
+
+// DefaultDNSResolver holds default values for DNSResolver.
+var DefaultDNSResolver = DNSResolver{
+	Port:     "4317",
+	Interval: 5 * time.Second,
+	Timeout:  1 * time.Second,
+}
+
+// SetToDefault implements river.Defaulter.
+func (args *DNSResolver) SetToDefault() {
+	*args = DefaultDNSResolver
+}
+
+func (dnsResolver *DNSResolver) Convert() loadbalancingexporter.DNSResolver {
+	return loadbalancingexporter.DNSResolver{
+		Hostname: dnsResolver.Hostname,
+		Port:     dnsResolver.Port,
+		Interval: dnsResolver.Interval,
+		Timeout:  dnsResolver.Timeout,
+	}
+}
+
+// Extensions implements exporter.Arguments.
+func (args Arguments) Extensions() map[otelconfig.ComponentID]otelcomponent.Extension {
+	//TODO: Do we need to set these extensions?
+	return args.Protocol.OTLP.Client.Extensions()
+}
+
+// Exporters implements exporter.Arguments.
+func (args Arguments) Exporters() map[otelconfig.DataType]map[otelconfig.ComponentID]otelcomponent.Exporter {
+	return nil
+}
+
+// GRPCClientArguments is the same as otelcol.GRPCClientArguments, but without an "endpoint" attribute
```

Comment:
I think this is fine since it is fundamentally different and we dont have a way to remove parameters.